core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleLiteral.java

```diff
@@ -49,12 +49,17 @@
 	 * The literal's language tag.
 	 */
 	private String language;
+	transient private Optional<String> optionalLanguage = Optional.empty();
```

Comment:
Because `getLanguage()` returns an optional and we often have to check for a language during sorting this ends up creating a lot of garbage through new Optional objects. Precomputing it is much faster.